pyvista/core/filters/uniform_grid.py

```diff
@@ -414,3 +414,265 @@ def image_threshold(
         # run the algorithm
         _update_alg(alg, progress_bar, 'Performing Image Thresholding')
         return _get_output(alg)
+
+    def fft(self, output_scalars_name='ImageScalars', progress_bar=False):
+        """Apply the fast fourier transform to the active scalars.
+
+        The input can have real or complex data in any components and data
+        types, but the output is always complex doubles, with the first
+        component containing real values and the second component containing
```

Comment:
I've cleaned this up with the implementation of complex datatypes.

Again, this could have been its own PR, but we would have needed a reason for it and implementing it here makes sense, but I'm open to splitting it up.

Reply:
Resolved in #2773